zip-0228.rst

```diff
@@ -0,0 +1,539 @@
+
+In the current (NU5) Zcash protocol, each Action includes a Spend Authorization Signature [#protocol-spendauthsig]_ that binds a specific spend instruction to a specific transaction and prevents replay attacks.
+However, in the context of Asset Swaps, there are two problems that prevent this mechanism from being carried over unchanged.
+
+- The SIGHASH represents a hash over a full consensus-compliant transaction object. Since a Swap Order is a trade intent to be used by the party performing the matching to form a valid bundle/transaction, the sender of the Swap Order cannot compute a SIGHASH as per ZIP244. This trader doesn't know all the transaction fields in advance, as they are set when the full bundle transaction is formed and sent to the chain.
+- The ``timeLimit`` parameter is a new addition to the protocol, and is therefore not included in the existing SIGHASH. Swap Orders need to be made non-malleable in order to ensure that the party performing the matching cannot change the time limit unilaterally.
+
+To make sure the time limit isn't malleable, we have the Order sender sign the Order content (i.e. the Action Group of the Order). 
+Thus, the sender “seals” the time limit within the Action Group, with the signature verification key being derived in the circuit. 
+This effectively links the time limit parameter to the Actions proof.
+A party trying to tweak the time limit of received orders while matching orders will then need to generate a new proof for the Actions in the Order, which isn't possible without knowing the full witness.
+
+This approach is natural and aligns with the goal of the protocol.
+Signing together all the contents of the Order prevents the "terms" of the Order from being modified.
+This ensures that the party performing the matching and settlement is only given the authority to match the Order with other Orders to create a valid transaction, and does not get custody of the Assets in the Order.
+
+Rationale for SIGHASH Computation Changes
+`````````````````````````````````````````
+
+The protocol specified in this ZIP makes certain changes to the transaction format, such as using the Action Groups level of abstraction, and adding a new ``timeLimit`` field. 
+The computation of SIGHASH is updated so as to capture all these changes into the hash, so as to bind all the parts of a transaction and make it non-malleable.
+
+Rationale for Binding Signature Changes
+```````````````````````````````````````
+
+One way in which a malicious matcher could exploit the protocol would be to break the atomicity of Swap Orders by only forming bundles with the Fee Actions of a Swap Order. 
+In this case, the matcher would relay on chain the “fee paying instructions” without executing the actual trade.
+
+By including :math:`\mathsf{bsk}` values in each Swap Order, and building the trasnaction :math:`\mathsf{bsk}` from these values as described in the `Binding Signature Changes`_ section, we prevent this malicious behavior.
```

Comment:
```suggestion
By including :math:`\mathsf{bsk}` values in each Swap Order, and building the transaction :math:`\mathsf{bsk}` from these values as described in the `Binding Signature Changes`_ section, we prevent this malicious behavior.
```

Reply:
Oops. Fixed this along with other smaller updates in [this commit](https://github.com/zcash/zips/pull/780/commits/69ff7a8dd7537dadc1c69031b1d44a66969b6962).